http_util.go

```diff
@@ -68,6 +68,9 @@ func newRequest(env Environment, method string, path string, body io.Reader, hea
 	if !strings.HasPrefix(path, "/") {
 		path = "/" + path
 	}
+	if env.Transport != nil {
```

Comment:
I would rather recommend to allow setting whole custom `httpClient`. For our use case it wouldn't matter much, but it has a fundamental flaw:
You're overwriting a package variable's field with an `Environment`-specific data. Therefore mutating the state for the whole library.

I recommend adding `Environment` to the `Do` method as second argument and `if custom httpClient, use that instead of a default one`. The `Do` method is called at two places, but both have access to `Environment` struct.

Reply:
`/http_request:13`
`/list_request.go:13`
`/request.go:20`